### PR TITLE
MOTECH-1974 Include object specific parameters as a part of the MotechEvent class

### PR DESCRIPTION
**Pull request by martokarski**

### CODE REVIEW

File: platform/event/src/main/java/org/motechproject/event/MotechEvent.java [hunk 1]
```diff
@@ -70,6 +73,60 @@ public void setId(UUID id) {
...
+     * Sets event as discarded
+     *
+     * @param value
+     */
+    public void setDiscarded(boolean value) {
+        discarded = value;
+    }
+
+    /**
+     * Returns whether event is broadcast
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/MotechEvent.java]: a broadcast event

File: platform/event/src/main/java/org/motechproject/event/MotechEvent.java [hunk 2]
```diff
@@ -70,6 +73,60 @@ public void setId(UUID id) {
...
+     *
+     * @return broadcast
+     */
+    public boolean isBroadcast() {
+        return broadcast;
+    }
+
+    /**
+     * Sets event as broadcast
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/MotechEvent.java]: a broadcast event

File: platform/event/src/main/java/org/motechproject/event/MotechEvent.java [hunk 3]
```diff
@@ -79,6 +136,24 @@ public String getSubject() {
     }
 
     /**
+     * Returns destination of the message
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/MotechEvent.java]: - the id of the Motech listener that this event is meant for

File: platform/event/src/main/java/org/motechproject/event/MotechEvent.java [hunk 4]
```diff
@@ -79,6 +136,24 @@ public String getSubject() {
...
+     *
+     * @return message destination
+     */
+    public String getMessageDestination() {
+        return messageDestination;
+    }
+
+    /**
+     * Sets message destination
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/MotechEvent.java]: - the id of the Motech listener that this event is meant for

File: platform/event/src/main/java/org/motechproject/event/listener/proxy/EventAnnotationBeanPostProcessor.java [hunk 5]
```diff
@@ -142,7 +142,7 @@ public void clearListeners(ApplicationContext applicationContext) {
         }
     }
 
-    //TODO:keeping required false for now, should be removed.
+    // @TODO:keeping required false for now, should be removed.
     @Autowired(required = false)
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/listener/proxy/EventAnnotationBeanPostProcessor.java]: why the @? btw, I think we should be good to remove required = false now anyway

**Comment by pgesek** [comment]
You will have to also create a commit for modules that removes the usage of the removed constants from the DHIS module.

File: platform/event/src/main/java/org/motechproject/event/MotechEvent.java [hunk 6]
```diff
@@ -15,14 +16,16 @@
  */
 public class MotechEvent implements Serializable {
     public static final String EVENT_TYPE_KEY_NAME = "eventType";
```
Comment (pgesek) [inline platform/event/src/main/java/org/motechproject/event/MotechEvent.java]: What's event type? Is this for the scheduler? If its only for the scheduler, we should move this to scheduler module, the scheduler service interface for example.

**Comment by sebbrudzinski** [comment]
LGTM

**Pull request merged**